### DISCUSSION TITLE
NimBLEDevice::init to take an max number of connection optional parameter

### DISCUSSION THREAD

**Discussion by davidlao2k** [h2zero/NimBLE-Arduino#810] [Ideas]
I use an esp32 to pull counters from 8 ble devices continuously. I often find myself having to manually updating nimconf.h with non-default CONFIG_BT_NIMBLE_MAX_CONNECTIONS, and easily forget that when working on different build environment or when a new version become available. 

Since this is just a pass through value, how about changing the init signature to below to make life easier?

bool NimBLEDevice::init(const std::string& deviceName, uint8_t maxConnections = CONFIG_BT_NIMBLE_MAX_CONNECTIONS)
...
   bt_cfg.ble_max_conn = maxConnections;
...

Any objection?

**Comment by h2zero** [comment]
The problem with this approach is that when using a device such as the nRF5x that the underlying stack statically allocates memory for the max connections, so this would be problematic and potentially cost a lot of ram for these devices without a need..
